tests/CMakeLists.txt

```diff
@@ -5,6 +5,8 @@
 # Moritz Scherer <scheremo@iis.ee.ethz.ch>
 
 add_subdirectory(generic)
-add_target_source(${TARGET_PLATFORM})
 
-# TODO: Add target-specific test directories
+# Check if folder has a CMakeLists.txt file
+if(EXISTS ${CMAKE_CURRENT_LIST_DIR}/${TARGET_PLATFORM}/CMakeLists.txt)
+  add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/${TARGET_PLATFORM})
+endif()
```

Comment:
This should probably fail loudly, or at least warn - maybe add an `else` with descriptive `FATAL_ERROR` or `WARNING`?